lib/node_modules/@stdlib/utils/parse-ndjson/lib/main.js

```diff
@@ -0,0 +1,81 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
+var isFunction = require( '@stdlib/assert/is-function' );
+var parseJSON = require( '@stdlib/utils/parse-json' );
+
+
+// MAIN //
+
+/**
+* Parses a string as newline-delimited JSON (NDJSON).
+*
+* @param {string} str - input string
+* @param {Function} [reviver] - transformation function applied to each line
+* @throws {TypeError} first argument must be a string
+* @throws {TypeError} reviver must be a function
+* @returns {Array<(*|Error)>} array of parsed values or parse errors
+*
+* @example
+* var obj = parseNDJSON( '{"name":"John"}\n{"name":"Doe"}' );
+* // returns [ {'name':'John'}, {'name':'Doe'} ]
+*/
+function parseNDJSON( str, reviver ) {
+	var parsed;
+	var lines;
+	var out;
+	var i;
+
+	if ( !isString( str ) ) {
+		throw new TypeError( 'invalid argument. First argument must be a string. Value: `%s`.', str );
```

Comment:
This is missing `format` for formatting the error message:

```suggestion
		throw new TypeError( format( 'invalid argument. First argument must be a string. Value: `%s`.', str ) );
```

Will need to require `var format = require( '@stdlib/string/format' );`.